server/action_creators.js

```diff
@@ -71,8 +71,12 @@ export function setOpen(orderId) {
   return dispatchUpdateOrder(orderId, Order.updateStatus(orderId, constants.OPEN));
 }
 
-export function setClosed(orderId, amounts) {
-  const transaction = Transaction.addTransaction(orderId, amounts)
+export function setClosed(orderId, transactionId, amounts) {
+  const addOrUpdateTransaction = transactionId ?
```

Comment:
Nice. I was debating whether the add and update transaction should be separated, and client would have to know which to call. That way we would avoid passing transactionId as undefined. But I think doing it this way is cleaner because the client just sends everything over, and the server knows what to do with it.